data-prepper-plugins/common/src/test/java/com/amazon/dataprepper/plugins/source/RandomStringSourceTests.java

```diff
@@ -30,15 +32,12 @@ public void testPutRecord() throws InterruptedException {
         final TestBuffer buffer = new TestBuffer(bufferQueue, 1);
         //Start source, and sleep for 100 millis
         randomStringSource.start(buffer);
-        Thread.sleep(500);
-        //Make sure that 1 record is in buffer
-        Assert.assertEquals(1, buffer.size());
+        Thread.sleep(1000);
```

Comment:
Alternative is to wrap L38 into a callback in awaitability like here: https://github.com/opensearch-project/data-prepper/blob/6365f38d9ec6ef810328bcf6cf61d6609d5fc219/data-prepper-plugins/otel-trace-raw-prepper/src/test/java/com/amazon/dataprepper/plugins/prepper/oteltrace/OTelTraceRawPrepperTest.java#L161-L168
so that you could set the wait time more conservatively without affecting the test case efficiency (if success). The tradeoff is introducing an extra test dependency

Reply:
Other projects in Data Prepper use awaitility. So adding a new dependency should not be a problem.

Great suggestion!